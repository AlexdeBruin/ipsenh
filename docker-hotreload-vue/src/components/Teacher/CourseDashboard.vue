<template>
  <div>
    <v-toolbar>
      <v-toolbar-side-icon class="icon">
        <v-icon color="white" large @click="$router.go(-1)">keyboard_arrow_left</v-icon>
      </v-toolbar-side-icon>
      <v-tabs grow v-model="tabs">
        <v-tab>Cijfers invoeren</v-tab>
        <v-tab>Cijfers inzien</v-tab>
      </v-tabs>
    </v-toolbar>

    <v-tabs-items v-model="tabs">
      <v-tab-item>
        <insert-grades/>
      </v-tab-item>
      <v-tab-item>
        <grades-overview/>
      </v-tab-item>
    </v-tabs-items>
  </div>
</template>

<script>
import InsertGrades from "./InsertGrades.vue";
import GradesOverview from "./GradesOverview.vue";

export default {
  data: () => ({
    tabs: null
  }),
  components: {
    "insert-grades": InsertGrades,
    "grades-overview": GradesOverview
  }
};
</script>

<style scoped>
.icon {
  float: left;
}
.v-toolbar {
  background-color: #3c3f5c !important;
  border-radius:10px;
  width:98.5%;
  margin-top: 8px !important;
  margin-left:auto;
  margin-right:auto;
}

.v-tabs__container,
.v-tabs__div {
    background-color: #3c3f5c !important;
}
</style>



