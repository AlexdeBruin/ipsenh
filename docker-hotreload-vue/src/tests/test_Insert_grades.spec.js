import CourseDashboard from '../components/Teacher/CourseDashboard.vue'
import {shallowMount} from '@vue/test-utils'

const wrapper = shallowMount(CourseDashboard)

describe('courseDashboard', () => {
    test('is a vue instance', () => {
        expect(wrapper.isVueInstance()).toBeTruthy()
    })
    it('contains a v-tab element', () => {
        expect(wrapper.contains('v-tab')).toBeTruthy()
    })
    test('renders correctly', () => {
        expect(wrapper.element).toMatchSnapshot()
    })
    it('tabs is null', () => {
        expect(wrapper.vm.tabs).toBe(null)
    })

})