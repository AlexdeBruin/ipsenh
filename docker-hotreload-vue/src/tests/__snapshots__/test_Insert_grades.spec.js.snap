// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`courseDashboard renders correctly 1`] = `
<div>
  <v-toolbar>
    <v-toolbar-side-icon
      class="icon"
    >
      <v-icon
        color="white"
        large=""
      >
        keyboard_arrow_left
      </v-icon>
    </v-toolbar-side-icon>
     
    <v-tabs
      grow=""
    >
      <v-tab>
        Cijfers invoeren
      </v-tab>
       
      <v-tab>
        Cijfers inzien
      </v-tab>
    </v-tabs>
  </v-toolbar>
   
  <v-tabs-items>
    <v-tab-item>
      <insert-grades-stub />
    </v-tab-item>
     
    <v-tab-item>
      <grades-overview-stub />
    </v-tab-item>
  </v-tabs-items>
</div>
`;
